src/a4/test/BoardSpaceTest.java

```diff
@@ -0,0 +1,94 @@
+package a4.test;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+
+import a4.domain.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class BoardSpaceTest {
+	BoardSpaceFactory test_factory = new BoardSpaceFactory();
```

Comment:
Do we need the factory in the boardSpace tests or should we just make a new board space? Would the factory have its own unit tests?